File: apartments.py
def f(k, a, b):
    a.sort()
    b.sort()

    i = 0
    j = 0
    n = len(a)
    m = len(b)

    count = 0
    while i < n and j < m:
        ai = a[i]
        bj = b[j]
        if bj - k <= ai <= bj + k:
            count += 1
            i += 1
            j += 1
        elif ai < bj - k:
            i += 1
        elif bj + k < ai:
            j += 1
    return count


def main():
    n, m, k = [int(x) for x in input().split()]
    a = [int(x) for x in input().split()]
    b = [int(x) for x in input().split()]

    print(f(k, a, b))


main()

File: Apartments.cpp
#include <math.h>

#include <algorithm>
#include <iostream>
#include <map>
#include <set>
#include <vector>

using namespace std;

typedef long long ll;

void solve() {
  size_t n, m;
  ll k;
  cin >> n >> m >> k;

  vector<ll> a;
  vector<ll> b;

  for (size_t i = 0; i < n; i++) {
    ll x;
    cin >> x;
    a.push_back(x);
  }

  for (size_t i = 0; i < m; i++) {
    ll x;
    cin >> x;
    b.push_back(x);
  }

  sort(a.begin(), a.end());
  sort(b.begin(), b.end());

  size_t i = 0;
  size_t j = 0;

  size_t count = 0;

  while (i < n && j < m) {
    ll ai, bj;
    ai = a[i];
    bj = b[j];

    if (bj - k <= ai && ai <= bj + k) {
      count++;
      i++;
      j++;
    } else if (ai < bj - k) {
      i++;
    } else if (bj + k < ai) {
      j++;
    }
  }
  cout << count << "\n";
}

int main() {
  ios::sync_with_stdio(0);
  cin.tie(0);
  solve();
  return 0;
}